pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractScalarTransformFunction.java

```diff
@@ -131,7 +131,7 @@ public TransformResultMetadata getResultMetadata() {
     final String[] stringValuesSV = _jsonFieldTransformFunction.transformToStringValuesSV(projectionBlock);
     final int[] results = new int[projectionBlock.getNumDocs()];
     for (int i = 0; i < results.length; i++) {
-      Object read = JsonPath.read(stringValuesSV[i], _jsonPath);
+      Object read = JsonPath.using(JSON_PARSER_CONFIG).parse(stringValuesSV[i]).read(_jsonPath);
```

Comment:
Shall we cache `JsonPath.using(JSON_PARSER_CONFIG)` as a constant? I think it is thread-safe

Reply:
I am a bit hesitant to do this since since it will allocate static memory for Parser, but if this function is never used in query, the parser will still take up memory space. Although, it seems like there is scope for both performance and memory improvement here. For example, currently `JsonExtractScalarFunction` object will be constructed fresh for each query that uses `json_extract_scalar` and if the query has multiple instances of `json_extract_scalar` then `JsonExtractScalarFunction` will be constructed multiple times. So one option may be to add a lazy function object cache to `TransformationFunctionFactory` to reuse function objects. Although, this is probably better handled in its own PR.

Reply:
It's just one single object on top of the Configuration which can be reused. I would not worry about the memory overhead of one single object. Also, it should never be initialized if this function is never used thus this class is not loaded.

Reply:
I have added the constant for now, although at some point in future it would be good to move towards having a function object cache in `TransformationFunctionFactory`.